Source/items.cpp

```diff
@@ -2677,59 +2696,57 @@ void CalcPlrItemVals(Player &player, bool loadgfx)
 	player._pDexterity = std::max(0, dadd + player._pBaseDex);
 	player._pVitality = std::max(0, vadd + player._pBaseVit);
 
-	if (player._pClass == HeroClass::Rogue) {
+	switch (pc) {
+	case HeroClass::Rogue:
 		player._pDamageMod = player._pLevel * (player._pStrength + player._pDexterity) / 200;
-	} else if (player._pClass == HeroClass::Monk) {
-		if (player.InvBody[INVLOC_HAND_LEFT]._itype != ItemType::Staff) {
-			if (player.InvBody[INVLOC_HAND_RIGHT]._itype != ItemType::Staff && (!player.InvBody[INVLOC_HAND_LEFT].isEmpty() || !player.InvBody[INVLOC_HAND_RIGHT].isEmpty())) {
-				player._pDamageMod = player._pLevel * (player._pStrength + player._pDexterity) / 300;
-			} else {
-				player._pDamageMod = player._pLevel * (player._pStrength + player._pDexterity) / 150;
-			}
-		} else {
+		break;
+	case HeroClass::Monk:
+		if (IsAnyOf(ItemType::Staff, leftHandItemType, rightHandItemType) || (leftHandEmpty && rightHandEmpty)) {
 			player._pDamageMod = player._pLevel * (player._pStrength + player._pDexterity) / 150;
+		} else {
+			player._pDamageMod = player._pLevel * (player._pStrength + player._pDexterity) / 300;
 		}
-	} else if (player._pClass == HeroClass::Bard) {
-		if (player.InvBody[INVLOC_HAND_LEFT]._itype == ItemType::Sword || player.InvBody[INVLOC_HAND_RIGHT]._itype == ItemType::Sword)
+		break;
+	case HeroClass::Bard:
+		if (IsAnyOf(ItemType::Sword, leftHandItemType, rightHandItemType)) {

```

Comment:
There are a lot of checks along the lines of "is the player holding a specific weapon in either hand", it'd be clearer to provide a function like `player.IsHoldingItem(ItemType)` where the definition looks like:

``` cpp
constexpr bool Player::IsHoldingItem(ItemType type) {
    return IsAnyOf(type, leftHandItem.type, rightHandItem.type);
}
```

Reply:
Also consider whether this function should check that the items are actually usable. Some of the uses do, some don't, is this inconsistency desirable?